pkg/model/bootstrapscript.go

```diff
@@ -72,6 +73,39 @@ func (b *BootstrapScript) ResourceNodeUp(ig *kops.InstanceGroup) (*fi.ResourceHo
 			return ""
 		},
 
+		"ProxyEnv": func() string {
+			scriptSnippet := ""
+
+			if ps != nil && ps.HTTPProxy.Host != "" {
+				httpProxyUrl := "http://"
+				if ps.HTTPProxy.User != "" {
+
+					httpProxyUrl += ps.HTTPProxy.User
+					if ps.HTTPProxy.Password != "" {
+						httpProxyUrl += "@" + ps.HTTPProxy.Password
+					}
+				}
+				httpProxyUrl += ps.HTTPProxy.Host + ":" + strconv.Itoa(ps.HTTPProxy.Port)
+				scriptSnippet =
+					"export http_proxy=" + httpProxyUrl + "\n" +
+						"export https_proxy=${http_proxy}\n" +
+						"export ftp_proxy=${http_proxy}\n" +
+						"export no_proxy=" + ps.ProxyExcludes + "\n" +
+						"echo \"export http_proxy=${http_proxy}\" >> /etc/default/docker\n" +
+						"echo \"export https_proxy=${http_proxy}\" >> /etc/default/docker\n" +
+						"echo \"export ftp_proxy=${http_proxy}\" >> /etc/default/docker\n" +
+						"echo \"export no_proxy=${no_proxy}\" >> /etc/default/docker\n" +
+						"echo \"export http_proxy=${http_proxy}\" >> /etc/environment\n" +
+						"echo \"export https_proxy=${http_proxy}\" >> /etc/environment\n" +
+						"echo \"export ftp_proxy=${http_proxy}\" >> /etc/environment\n" +
+						"echo \"export no_proxy=${no_proxy}\" >> /etc/environment\n" +
+						"echo DefaultEnvironment=\\\"http_proxy=${http_proxy}\\\" \\\"https_proxy=${http_proxy}\\\" \\\"ftp_proxy=${http_proxy}\\\" \\\"no_proxy=${no_proxy}\\\" >> /etc/systemd/system.conf\n" +
+						"systemctl daemon-reload\n" +
+						"systemctl daemon-reexec\n" +
+						"echo \"Acquire::http::Proxy \\\"${http_proxy}\\\";\" > /etc/apt/apt.conf.d/30proxy\n\n"
```

Comment:
Oh, we need to support more than just Debian.  `kops` supports RHEL, Centos, Ubuntu, CoreOS, and Debian.  We are probably going to need to move this into nodeup.  We do not check the OS type when executing this, and this may be getting messy enough that we may need nodeup to do this.  Let me sleep on it, and can you ping me tomorrow via slack?

Bottom line we need to support apt, yum and whatever the heck CoreOS uses.